packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  Component,
+  Element,
+  Host,
+  Method,
+  Prop,
+  State,
+  h,
+} from "@stencil/core";
+import {
+  IcBadgePositions,
+  IcBadgeTypes,
+  IcBadgeVariants,
+} from "./ic-badge.types";
+import {
+  IcColorRGBA,
+  IcSizes,
+  IcThemeForeground,
+  IcThemeForegroundEnum,
+} from "../../utils/types";
+import {
+  getCssProperty,
+  getParentElement,
+  hexToRgba,
+  rgbaStrToObj,
+} from "../../utils/helpers";
+
+@Component({
+  tag: "ic-badge",
+  styleUrl: "ic-badge.css",
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class Badge {
+  private customColorRGBA: IcColorRGBA;
+  private foregroundColour: IcThemeForeground;
+
+  @Element() el: HTMLIcBadgeElement;
+
+  @State() visible: boolean = true;
+
+  /**
+   * The accessible label of the badge component to provide context for screen reader users.
+   */
+  @Prop() accessibleLabel?: string;
```

Comment:
This causes the screen reader to read out "badge displaying undefined" if it's not provided - should it be a required prop?